nixos/modules/services/web-apps/shiori.nix

```diff
@@ -31,6 +31,20 @@ in {
         example = "/shiori";
         description = "The root of the Shiori web application";
       };
+
+      httpSecretKey = mkOption {
+        type = types.str;
+        example = "SuperSecretPassword";
+        description =
+          "When empty all sessions will be invalidated on server restart";
+      };
```

Comment:
Do you think you could make this take a file instead, and inject the contents into the environment variable on startup? It's not much more secure, but at least you won't have to leave the key in the nix store.

Also, it's probably better to represent *unset* as `null` instead of empty string.

Reply:
I changed httpSecretKey in favor for environmentFile, this way the user can not only set up this environment key variable themselves (which is optional) but also, whichever other keys they decide to, linked to the documentation on the description, and implemented your null as unset suggestion! for everything except listen address, that one has to be "" for the default to trigger.